src/no-side-effects.spec.ts

```diff
@@ -176,37 +294,125 @@ createTester().run(ruleId, rule, {
         { messageId: 'NO_SIDE_EFFECTS' },
         { messageId: 'NO_SIDE_EFFECTS' },
         { messageId: 'NO_SIDE_EFFECTS' },
-        { messageId: 'NO_SIDE_EFFECTS' },
       ],
-      options: [{ excludedIdentifiers: ['assert', 'debug', 'log', 'Symbol.for'] }],
+      options: [{ excludedIdentifiers: ['debug', 'log', 'Symbol.for', 'Object.freeze'] }],
       name: 'Invalid case with multiple side effects including export default declarations and async operations',
     },
     {
       code: `import { strict as assert } from 'node:assert';
-                import http from 'node:http';
-                import debug from 'debug';
-                
-                import { getConfiguration } from 'test';
-                                
-                const unresolvedConfiguration = getConfiguration(root());
-                const configuration = await service(unresolvedConfiguration);
-                const symbol1 = Symbol.for('foo');
-                const server = http.createServer(logger(configuration.requestHandler, unresolvedConfiguration.name));
-                server.listen(Number.parseInt(unresolvedConfiguration.env['PORT'], 10), '0.0.0.0');
-                
-                await new Promise((resolve) => {
-                  server.on('listening', resolve);
-                });
-                export * from './module';`,
+              import http from 'node:http';
+              import debug from 'debug';
+              
+              import { getConfiguration } from 'test';
+                              
+              const unresolvedConfiguration = getConfiguration(root());
+              const configuration = await service(unresolvedConfiguration);
+              const symbol1 = Symbol.for('foo');
+              
+               const object = {
+                  prop: 'foo'
+              };
+             
+              assert.ok(statusCode === StatusCodes.OK, 'Status code is not OK');
+
+              Object.freeze(object);
+              
+              try {
+                 object.prop = 'bar';
+              } catch (error) {
+                  console.error(error); 
+              }
+              
+              Symbol.for('foo');
+              export default obj;
+              const server = http.createServer(logger(configuration.requestHandler, unresolvedConfiguration.name));
+              server.listen(Number.parseInt(unresolvedConfiguration.env['PORT'], 10), '0.0.0.0');
+              
+              await new Promise((resolve) => {
+                server.on('listening', resolve);
+              });
+              export * from './module';`,
       errors: [
         { messageId: 'NO_SIDE_EFFECTS' },
```

Comment:
in general, simplify test case to focus on one error at the time can make the test easier to understand and related to the implementation logic. is it possible to refactor and remove some redundancy across the test cases.